File: generate_screenshot.py
#!/usr/bin/env python3
"""
使用Playwright生成HTML页面长图
"""
import asyncio
import os
from playwright.async_api import async_playwright

async def generate_screenshot():
    html_path = os.path.abspath("Ignite X&Y 创新机制升级.html")
    output_path = os.path.abspath("Ignite创新机制介绍-长图.png")
    
    print("🚀 正在启动浏览器...")
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()
        
        # 设置手机屏幕宽度（iPhone 13 Pro Max）
        await page.set_viewport_size({"width": 428, "height": 926})
        
        print(f"📄 正在加载HTML文件: {html_path}")
        await page.goto(f"file://{html_path}", wait_until="networkidle")
        
        # 等待页面渲染完成
        await page.wait_for_timeout(1000)
        
        print("📸 正在生成长图...")
        # 截取整个页面（全长图）
        await page.screenshot(
            path=output_path,
            full_page=True,
            type="png"
        )
        
        print(f"✅ 长图已生成: {output_path}")
        await browser.close()

if __name__ == "__main__":
    os.chdir("/Users/hoimanszeto/.openclaw/workspace")
    asyncio.run(generate_screenshot())

File: generate-screenshot.js
const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

(async () => {
  const htmlPath = path.join(__dirname, 'Ignite创新机制介绍.html');
  const outputPath = path.join(__dirname, 'Ignite创新机制介绍-长图.png');

  console.log('正在启动浏览器...');
  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });

  const page = await browser.newPage();
  
  // 设置移动端视口（手机宽度）
  await page.setViewport({
    width: 750,
    height: 1334,
    deviceScaleFactor: 2
  });

  console.log('正在加载HTML文件...');
  await page.goto(`file://${htmlPath}`, {
    waitUntil: 'networkidle0'
  });

  console.log('正在生成长图...');
  // 获取页面完整高度
  const bodyHandle = await page.$('body');
  const { height } = await bodyHandle.boundingBox();
  await bodyHandle.dispose();

  // 设置视口高度为整个页面高度
  await page.setViewport({
    width: 750,
    height: Math.ceil(height),
    deviceScaleFactor: 2
  });

  // 截取整个页面
  await page.screenshot({
    path: outputPath,
    fullPage: true,
    type: 'png'
  });

  console.log(`✅ 长图已生成: ${outputPath}`);
  
  await browser.close();
})();
